internal/hardware/versioned-requirements_test.go

```diff
@@ -0,0 +1,163 @@
+package hardware
+
+import (
+	"os"
+
+	"github.com/kelseyhightower/envconfig"
+	. "github.com/onsi/ginkgo"
+	"github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
+	"github.com/openshift/assisted-service/models"
+	"github.com/openshift/assisted-service/pkg/conversions"
+)
+
+const (
+	requirementsEnv = "HW_VALIDATOR_REQUIREMENTS"
+)
+
+var _ = Describe("Versioned Requirements", func() {
+	BeforeEach(func() {
+		err := os.Unsetenv(requirementsEnv)
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	It("should be set to default when no env variable", func() {
+		cfg := ValidatorCfg{}
+
+		err := envconfig.Process("", &cfg)
+
+		Expect(err).ToNot(HaveOccurred())
+		Expect(cfg.VersionedRequirements).To(BeEmpty())
+	})
+
+	table.DescribeTable("should be decoded from JSON", func(json string, expected map[string]models.VersionedHostRequirements) {
+		_ = os.Setenv(requirementsEnv, json)
+		cfg := ValidatorCfg{}
+
+		err := envconfig.Process("", &cfg)
+
+		Expect(err).ToNot(HaveOccurred())
+		Expect(cfg.VersionedRequirements).To(BeEquivalentTo(expected))
+	},
+		table.Entry("empty", "[]", map[string]models.VersionedHostRequirements{}),
+		table.Entry("One version - full",
+			`[{
+  			"version": "4.6.0",
+  			"master": {
+  			  "cpu_cores": 4,
+  			  "ram_mib": 16384,
+  			  "disk_size_gb": 120,
+ 			  "installation_disk_speed_threshold_ms": 2
+  			},
+  			"worker": {
+  			  "cpu_cores": 2,
+  			  "ram_mib": 8192,
+  			  "disk_size_gb": 120,
+		  	  "installation_disk_speed_threshold_ms": 3
+  			}
+		}]`,
+			map[string]models.VersionedHostRequirements{
+				"4.6.0": {
+					Version: "4.6.0",
+					MasterRequirements: &models.ClusterHostRequirementsDetails{CPUCores: 4, DiskSizeGb: 120,
+						RAMMib: conversions.GibToMib(16), InstallationDiskSpeedThresholdMs: 2},
+					WorkerRequirements: &models.ClusterHostRequirementsDetails{CPUCores: 2, DiskSizeGb: 120,
+						RAMMib: conversions.GibToMib(8), InstallationDiskSpeedThresholdMs: 3},
+				}}),
+		table.Entry("One version - sparse",
+			`[{
+  			"version": "4.6.0",
+  			"master": {
+  			  "ram_mib": 16384,
+  			  "disk_size_gb": 120,
+			  "installation_disk_speed_threshold_ms": 3
+  			},
+  			"worker": {
+  			  "cpu_cores": 2,
+  			  "disk_size_gb": 120,
+			  "installation_disk_speed_threshold_ms": 2
+  			}
+		}]`,
+			map[string]models.VersionedHostRequirements{
+				"4.6.0": {
+					Version: "4.6.0",
+					MasterRequirements: &models.ClusterHostRequirementsDetails{CPUCores: 0, DiskSizeGb: 120,
```

Comment:
Minimum zero CPUs? Shouldn't it be invalid?

Reply:
You're right; RAM, disk and CPU should be greater than 0 if requirements for node type are present. Do you think it should be a fatal error that kills the service or would logging alone be sufficient?

Reply:
I think it should be fatal, otherwise it would just get ignored in all of the other logs mess